src/src_main/crypto.c

```diff
@@ -328,6 +328,31 @@ uint16_t Crypto_Calc_CRC16(uint8_t *data, int size)
     return crc;
 }
 
+/*set parameters for an encrypted TLS connection*/
+int32_t Crypto_Config_MariaDB_TLS_Connection(char* mysql_username, char* mysql_password, char* mysql_hostname, char* mysql_database, uint16_t mysql_port, uint8_t encrypted_connection, char* ssl_cert, char* ssl_key, char* ssl_ca, char* ssl_capath,char* bind_address)
```

Comment:
By creating your own separate function specifically for TLS connections, you are offloading to the client applications the logic of choosing user-pass/tls/mtls connections ... That's not what we want... they will just parse whatever properties their configs have and pass them upstream. Combine this function into the Config MariaDB function and just document which parameters are required for each mode in a comment.

Reply:
There should be an enum added to crypto_config_structs.h that specify each of the supported options.

Reply:
Crypto_Config_MariaDB_TLS_Connection () is removed & the parameters required for an encrypted connection are added to Crypto_Config_MariaDB()